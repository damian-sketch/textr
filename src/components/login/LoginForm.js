import { useState } from 'react'
import axios from 'axios'
import './styles.css';

const LoginForm = () => {
    const projectId = process.env.REACT_APP_PROJECT_ID

    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
    const [error, setError] = useState('');

    const handleSubmit = async (e) => {
        e.preventDefault();

        //header for authentication
        const authObject = { 'Project-ID': projectId, 'User-Name': username, 'User-Secret': password}
        
        // store credentials in local storage if they are valid
        try {
            await axios.get('https://api.chatengine.io/chats', { headers: authObject });
            localStorage.setItem('username', username)
            localStorage.setItem('password', password)

            window.location.reload()
         } catch (error) {
             setError('Incorrect credentials, try again')
         }
                            
    }

    


    return(
        <div className="wrapper">
            <div className="form">
                <h1 className="title">TEXTR</h1>
                <form onSubmit={handleSubmit}>
                    <input type="text" value={username} onChange={(e) => setUsername(e.target.value)} className="input" placeholder="Username" required/>
                    <input type="password" value={password} onChange={(e) => setPassword(e.target.value)} className="input" placeholder="Password" required/>
                    <div align="center">
                        <button type="submit" className="button">
                            <span>Login</span>
                        </button>
                    </div>
                    <h2 className="error"> {error} </h2>
                </form>
            </div>
        </div>
    )

}

export default LoginForm;