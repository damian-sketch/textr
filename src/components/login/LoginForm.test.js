import {render, cleanup} from '@testing-library/react'
import LoginForm from './LoginForm'
import '@testing-library/jest-dom'

//cleanup to prevent memory leaks
afterEach(cleanup)

// test case
it('should verify login button is displayed', () => {
    const { getByTestId } = render(<LoginForm />)

    //make sure Login button containd Login text(this means form is displayed)
    expect(getByTestId('login-button')).toHaveTextContent('Login')
})
