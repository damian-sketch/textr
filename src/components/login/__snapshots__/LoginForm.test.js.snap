// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="form"
    >
      <h1
        class="title"
      >
        TEXTR
      </h1>
      <form>
        <input
          class="input"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
        <input
          class="input"
          placeholder="Password"
          required=""
          type="password"
          value=""
        />
        <div
          align="center"
        >
          <button
            class="button"
            type="submit"
          >
            <span>
              Login
            </span>
          </button>
        </div>
        <h2
          class="error"
        >
            
        </h2>
      </form>
      <button>
        Sign up
      </button>
    </div>
  </div>
</DocumentFragment>
`;
